### PR TITLE
closes #16 backend research. 

### LINKED ISSUE #16: Backend research
Find an appropriate backend technology.

Must have:
- Reactive data
- Web API access with JSON
- Authentication
- Authorization (Maybe players, coaches, consultants)

Nice to have:
- Implementation language as C#, JavaScript or TypeScript

Is it possible to use Meteor as NativeScript backend?

### PR DESCRIPTION
**Pull request by Karnich**
If you have anything else to add to the conclusion let me know :)

### CODE REVIEW

File: Master.tex [hunk 1]
```diff
@@ -15,6 +15,10 @@
 
 \input{files/preStudies/preStudies}
 
+\input{files/preStudies/frameworks}
+
+\input{files/preStudies/backends}
```
Comment (Crevil) [inline Master.tex]: Include in `files/preStudies/preStudies` instead.

File: files/preStudies/backends.tex [hunk 2]
```diff
@@ -0,0 +1,58 @@
+\section{Backends}
+
+This section will evaluate different options to support the project with data storage and possibly for running heavy jobs that you do not what the clients to run. Some \textit{must have} and \textit{nice to have} capabilities will be listed below.
```
Comment (Crevil) [inline files/preStudies/backends.tex]: will be listed -> is listed

File: files/preStudies/backends.tex [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	
+\textbf{Must have:}
+	\begin{itemize}
+	\item Be able to store data
+	\item Support authentication.
```
Comment (Crevil) [inline files/preStudies/backends.tex:8]: End with dot or not. Be consistent. Same for next list.

File: files/preStudies/backends.tex [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	\item Generic API
+	\item Reactive data. Updated data in the database should reflect on the client.
+	\end{itemize}
+	
+
+\subsubsection{Firebase}
+
+\textbf{Function:} Data storage
+\\
+\textbf{Database type:} NoSQL (JSON)
```
Comment (Crevil) [inline files/preStudies/backends.tex]: JSON as database type?

File: files/preStudies/backends.tex [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	
+	\begin{itemize}
+	\item Generic API
+	\item Reactive data. Updated data in the database should reflect on the client.
+	\end{itemize}
+	
+
+\subsubsection{Firebase}
+
+\textbf{Function:} Data storage
```
Comment (Crevil) [inline files/preStudies/backends.tex]: Is this not Reactive Data Storage as Apollo?

File: files/preStudies/backends.tex [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+\\
+\textbf{Database type:} NoSQL (JSON)
+\\
+\textbf{Description:}
+\\
+Firebase is a database platform from \textit{Google}. It has a lot of features you do not normally see on a database. One of the important things is authorization. Usually stuff like authorization goes through some sort of Web API before access is allowed to the database. Firebase can handle this aspect without a Web API to parent it. Firebase is a real-time database, which means that the data is reactive. So whenever there is a change in the database, that change is pushed to the clients to keep them up to date at all times. Pushing data into the database is done with JSON. When using Firebase you can not run functions or jobs like in other systems. But a company named \textit{Zapier} have made this available through a product of theirs. This is an extra cost and might not be needed, But it is nice to know that there is an option for this.
```
Comment (Crevil) [inline files/preStudies/backends.tex]: Firebase in italics all over.
Can you rephrase the first three sentences to not all start with Firebase ... 
other systems. But a -> other systems, but 
I think you could skip the last sentence, as it's kind of redundant?

File: files/preStudies/backends.tex [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+
+\subsubsection{Apollo}
+
+\textbf{Function:} Reactive Data storage
+\\
+\textbf{Database type:} None specific, free to chooose. Client application interface is GraphQL.
```
Comment (Crevil) [inline files/preStudies/backends.tex]: GraphQL in italics

File: files/preStudies/backends.tex [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+\\
+\textbf{Description:}
+\\
+Apollo or the Apollostack is a modern data stack that offers reactive data. This means that your data will always be up-to-date, if you want it to be that is. You can choose your poise. If you do not need reactive data you can choose not to. Apollo is a data stack provided by the \textit{Meteor} team. As opposed to \textit{Meteor}, Apollo is ment to be useable in any application. There are integration guides for Angular 2, Redux, Meteor, React and React native. But you should be able to integrate it into any javascript frontend. Apollo uses GraphQL, which consists of 2 parts. A Schema part that tells what type of data is available on the server and a query language that enabled the client to describe the data it needs. This means that the client have no idea what type of database is behind GraphQL. It is comparable to REST, but where you very clearly define what you need instead of getting an object like you usually would when using REST.
```
Comment (Crevil) [inline files/preStudies/backends.tex]: Apollo in italics all over.
it to be that is -> it to be, that is.
Angular2, Redux, Metoer, React, React Native in italics.
No dot before the "but"s.
javascript -> JavaScript
GraphQL in italics.
enabed -> enables
The last phrase is a bit pleonastic. Maybe just mention that you define what properties you want as compared to a typical REST.

File: files/preStudies/backends.tex [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+	\textbf{Support authentication} & \cmark & \cmark \\ 
+	\hline 
+	Generic API & \xmark & \cmark \\ 
+	\hline 
+	Reactive data & \cmark & \cmark \\ 
+	\hline 	
+	\end{tabularx} 
+	
+\subsubsection{Conclusion}
+\textit{Apollo} fulfills all of our requirements and appeals to us for the fact that the client and the server gets seperated and independant of each other. With Firebase the client application would be bound to it. Apollo let's us pick what database we want and we can even have several and different databases without the client knowing about it. It provides us with flexibility and seperates the front and the backend, at the cost of more work. The selection on what backend to use therefore falls upon \textit{Apollo}.
```
Comment (Crevil) [inline files/preStudies/backends.tex]: seperated and independant -> seperated and are independant
Firebase and Apollo in italics.
let's -> lets

**Comment by Crevil** [comment]
GJ!!

File: files/preStudies/backends.tex (hunk 8 above)
Comment (Karnich) [inline files/preStudies/backends.tex]: how is this instead: 
It is comparable to REST, but with \textit{GraphQL} you define what properties you need. This improves data usage cause you do not send unneeded information over the wire.

Comment (Crevil) [inline files/preStudies/backends.tex]: I like it!
